test/should_fail/shortcut_ops.erl

```diff
@@ -0,0 +1,16 @@
+-module(shortcut_ops).
+
+-compile(export_all).
+
+-spec andalso_too_precise1(true, number()) -> number().
+andalso_too_precise1(True, N) -> True andalso N.
+
+-spec andalso_too_precise2(false, number()) -> false.
```

Comment:
For `should_fail`, we only require one function to fail in a module. You probably want all these to fail. (We can perhaps change how these tests work, but sometimes you want a (passing) helper function in a test module like this, so requiring all functions to fail is maybe not good either. Perhaps we can require all the exported functions to fail...)

Reply:
I was hoping that after #59 gets merged we could work on checking that we get the right kind and number of errors from each `should_fail` and didn't want to spend work on splitting them up just to merge them back together again.